build_runner_core/lib/src/generate/build_target.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+class BuildTarget {
```

Comment:
I didn't realize this was in build_runner_core (I thought it was the daemon?). That makes me less comfortable about the duplicate name with build_config...

Reply:
I'm open to changing it. I do really like BuildTarget as it is very descriptive even if the term is overloaded. What about `BuildDestination`?

Reply:
@natebosch thoughts?

Reply:
`BuildDirectory` would be consistent with `buildDirs`...

Reply:
Noted. I'll update the PR in a few.